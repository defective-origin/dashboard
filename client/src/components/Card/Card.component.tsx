import React, { forwardRef } from 'react'
import MuiCard from '@mui/material/Card'
import MuiCardActionArea from '@mui/material/CardActionArea'
import MuiCardActions from '@mui/material/CardActions'
import MuiCardContent from '@mui/material/CardContent'
import MuiCardHeader from '@mui/material/CardHeader'
import MuiCardMedia from '@mui/material/CardMedia'

// ---| core |---
import { cn } from 'tools'

// ---| components |---
import Divider from 'components/Divider'

// ---| self |---
import css from './Card.module.scss'

export type CardProps = {
  horizontal?: boolean
  divided?: boolean
  style?: React.CSSProperties
  className?: string
  children?: React.ReactNode
}

// TODO: remove forwardRef after migrating to react 19

/**
 * Component description.
 *
 * How to use
 * @example
 * <Card />
 */
export const Card = forwardRef<HTMLDivElement, CardProps>((props, ref): JSX.Element | null => {
  const { divided, horizontal, children, className, ...otherProps } = props
  const _className = cn(css.Card, {
    [css.horizontal]: horizontal,
    [css.divided]: divided,
  }, className)

  return (
    <MuiCard ref={ref} className={_className} {...otherProps}>
      {children}
    </MuiCard>
  )
}) as unknown as React.FC<CardProps> & Record<'ActionArea' | 'Actions' | 'Content' | 'Header' | 'Media' | 'Divider', React.FC<any>>

Card.displayName = 'Card'

Card.ActionArea = MuiCardActionArea
Card.Actions = MuiCardActions
Card.Content = MuiCardContent
Card.Header = MuiCardHeader
Card.Media = MuiCardMedia
Card.Divider = Divider

export default Card
