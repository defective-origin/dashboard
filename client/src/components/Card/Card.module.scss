@import 'src/App/App.module';

:global {
  .dark {
  }

  .light {
  }
}

.Card {
  display: flex;
  flex-direction: column;
}

.horizontal {
  flex-direction: row;
}

.divided {
  border-bottom: 1px solid var(--divider-color);

  &.horizontal > * {
    border-right: 1px solid var(--divider-color);
  }

  &.Vertical > :last-child,
  &.horizontal > :last-child {
    border-right: none;
    border-bottom: none;
  }
}
