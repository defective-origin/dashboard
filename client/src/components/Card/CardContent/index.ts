export { default } from './CardContent.component'
export * from './CardContent.component'
