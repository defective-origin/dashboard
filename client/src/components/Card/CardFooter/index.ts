export { default } from './CardFooter.component'
export * from './CardFooter.component'
