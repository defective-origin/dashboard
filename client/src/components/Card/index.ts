export { default } from './Card.component'
export * from './Card.component'
