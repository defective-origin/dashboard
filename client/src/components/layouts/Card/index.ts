/* INJECT_IMPORT_PLACE */

/* INJECT_DEFAULT_EXPORT_PLACE */
export { default } from './Card.component'

/* INJECT_EXPORT_PLACE */
export * from './Card.component'
