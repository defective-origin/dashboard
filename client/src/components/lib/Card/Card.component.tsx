import React from 'react'
import MuiCard, { CardProps as MuiCardProps } from '@mui/material/Card'
import MuiCardActionArea from '@mui/material/CardActionArea'
import MuiCardActions from '@mui/material/CardActions'
import MuiCardContent from '@mui/material/CardContent'
import MuiCardHeader from '@mui/material/CardHeader'
import MuiCardMedia from '@mui/material/CardMedia'

// ---| components |---
import { Divider } from 'components/lib/Divider'

// ---| common |---
import { cn } from 'common/tools'

// ---| self |---
import css from './Card.module.scss'

export type CardProps = MuiCardProps & {
  className?: string
  children?: React.ReactNode
  horizontal?: boolean
  divided?: boolean
}

/**
 * Component description.
 *
 * How to use
 * @example
 * <Card />
 */
export function Card(props: CardProps): JSX.Element {
  const { divided, horizontal, children, className, ...otherProps } = props
  const _className = cn(
    css.Card,
    horizontal && css.Horizontal,
    divided && css.Divided,
    className,
  )

  return <MuiCard className={_className} {...otherProps}>{children}</MuiCard>
}

Card.displayName = 'Card'

Card.ActionArea = MuiCardActionArea
Card.Actions = MuiCardActions
Card.Content = MuiCardContent
Card.Header = MuiCardHeader
Card.Media = MuiCardMedia
Card.Divider = Divider

export default Card
