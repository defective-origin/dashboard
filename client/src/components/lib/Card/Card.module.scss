@import 'src/App/App.module';

:global {
  .dark {
  }

  .light {
  }
}

.Card {
  display: flex;
  flex-direction: column;
}

.Horizontal {
  flex-direction: row;
}

.Divided {
  border-bottom: 1px solid var(--divider-color);

  &.Horizontal > * {
    border-right: 1px solid var(--divider-color);
  }

  &.Vertical > :last-child,
  &.Horizontal > :last-child {
    border-right: none;
    border-bottom: none;
  }
}
