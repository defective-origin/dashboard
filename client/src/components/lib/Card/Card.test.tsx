// ---| tests |---
import { render } from 'tests'

// ---| self |---
import Card from './Card.component'

describe('[Card] component', () => {
  it('should render component', () => {
    const container = render(<Card />)

    expect(container.snapshot()).toMatchSnapshot()
  })
})
