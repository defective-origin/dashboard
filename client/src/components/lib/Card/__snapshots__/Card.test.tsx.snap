// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`[Card] component > should render component 1`] = `<DocumentFragment />`;
